src/main/java/makaroshyna/onlinebookstore/repository/book/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,24 @@
+package makaroshyna.onlinebookstore.repository.book;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import makaroshyna.onlinebookstore.model.Book;
+import makaroshyna.onlinebookstore.repository.SpecificationProvider;
+import makaroshyna.onlinebookstore.repository.SpecificationProviderManager;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
```

Comment:
we can just operate our enum instead of string. We should always do that if we have such an opportunity, that would be more type-safe
```suggestion
    public SpecificationProvider<Book> getSpecificationProvider(BookParameterName key) {
```